plotly/tools.py

```diff
@@ -1559,38 +1551,43 @@ def _trisurf(x, y, z, simplices, colormap=None, dist_func=None,
 
         min_mean_dists = np.min(mean_dists)
         max_mean_dists = np.max(mean_dists)
-        facecolor = ([FigureFactory._map_z2color(zz, colormap, min_mean_dists,
-                      max_mean_dists) for zz in mean_dists])
-        ii, jj, kk = FigureFactory._tri_indices(simplices)
+        facecolor = FigureFactory._map_z2color(mean_dists, colormap,
```

Comment:
This seems to be the way that trisurf defines color of the faces, no? It looks here like it will expect `dist_func` to be callable on an `x, y, z` tuple, so don't we want to support someone giving an array of shape (n_triangles,) that just has rgb strings in it? Alternatively, if it's relatively easy to change the colors after the data has been created, maybe it's better to have a different high level function that changes them post-hoc, so that we don't re-do the triangle computations each time the colors change. In either case, I'd vote for a different variable name than `dist_func`...maybe `color_func` or just `colors`, and then explain in the docs that it could be a function. WDYT?